### ISSUE TITLE
Rustc should warn, not fail on bad doc comments

### ISSUE THREAD

**Issue by Manishearth** [rust-lang/rust#18444]
[source](http://www.reddit.com/r/rust/comments/2kqhnz/is_there_a_standard_way_to_document_parameters_in/)

The following MWE fails to compile with `error: expected item after doc comment`

```
fn main(){
}

/// Hi
```

this one fails to compile with a generic `unexpected token: /// some input`

```
fn main(){

}

fn foo(
  /// some input
  x: u8
  ) {

}
```

Comments should be ignored by the compiler, even doc comments. It makes sense to sanity-check comments during compilation, but can this be a warning instead of an error? In general, the `///` token should be treated as a comment after perhaps linting for doc comment correctness.

**Comment by steveklabnik** [comment]
This is a language semantic and so would need an RFC to change. Please persue one if you still feel strongly abou tthis.
